src/Microsoft.VisualStudio.LanguageServices.Razor/BackgroundDocumentGenerator.cs

```diff
@@ -257,23 +284,35 @@ private void ProjectManager_Changed(object sender, ProjectChangeEventArgs e)
                     }
 
                 case ProjectChangeKind.DocumentAdded:
+                case ProjectChangeKind.DocumentChanged:
                     {
-                        var project = _projectManager.GetLoadedProject(e.ProjectFilePath);
-                        Enqueue(project, project.GetDocument(e.DocumentFilePath));
+                        var project = e.Newer;
+                        var document = project.GetDocument(e.DocumentFilePath);
+
+                        Enqueue(project, document);
+                        foreach (var relatedDocument in project.GetRelatedDocuments(document))
+                        {
+                            Enqueue(project, document);
+                        }
```

Comment:
This is the next important part. When a document changes, we can just enqueue a change to all related documents - which will have already had their state reset.

Reply:
Ya it's awesome to see it all come together. 🥂